File: dynamicpdf-csharp-examples/Examples/SectionsExample.cs

using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements;

namespace DynamicPDFCoreSuite.Examples
{
    class SectionsExample
    {
        public static void Run()
        {
            Document document = new Document();
            Template template = new Template();
            template.Elements.Add(new PageNumberingLabel("%%SP%% of %%ST%%", 0, 680, 512, 12, Font.Helvetica, 32, TextAlign.Center));
            document.Sections.Begin(NumberingStyle.Numeric, template);
            document.Pages.Add(new Page());
            document.Sections.Begin(NumberingStyle.AlphabeticUpperCase, template);
            document.Pages.Add(new Page());
            document.Pages.Add(new Page());
            document.Pages.Add(new Page());
            document.Sections.Begin(NumberingStyle.RomanUpperCase, template);
            document.Pages.Add(new Page());
            document.Pages.Add(new Page());
            document.Pages.Add(new Page());
            document.Pages[3].ApplySectionTemplate = false;
            document.Draw(Util.GetPath("Output/sections-output.pdf"));
        }

    }
}

File: dynamicpdf-vbnet-examples/Examples/SectionsExample.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements

Namespace DynamicPDFCoreSuite.Examples
    Class SectionsExample
        Public Shared Sub Run()
            Dim document As New Document()
            Dim template As New Template()
            template.Elements.Add(New PageNumberingLabel("%%SP%% of %%ST%%", 0, 680, 512, 12, Font.Helvetica, 12, TextAlign.Center))

            document.Sections.Begin(NumberingStyle.RomanLowerCase)
            document.Pages.Add(New Page()) ' Page 1
            document.Pages.Add(New Page()) ' Page 2

            document.Sections.Begin(NumberingStyle.Numeric, template)
            document.Pages.Add(New Page()) ' Page 3
            document.Pages.Add(New Page()) ' Page 4
            document.Pages.Add(New Page()) ' Page 5

            document.Sections.Begin(NumberingStyle.RomanLowerCase, "Appendix A - ")
            document.Pages.Add(New Page()) ' Page 6
            document.Pages.Add(New Page()) ' Page 7

            document.Draw(Util.GetPath("Output/sections-output.pdf"))
        End Sub
    End Class
End Namespace

